src/main/java/mate/academy/springboot/datajpa/service/impl/ProductServiceImpl.java

```diff
@@ -0,0 +1,47 @@
+package mate.academy.springboot.datajpa.service.impl;
+
+import java.util.List;
+import java.util.NoSuchElementException;
+import mate.academy.springboot.datajpa.model.Product;
+import mate.academy.springboot.datajpa.repository.ProductRepository;
+import mate.academy.springboot.datajpa.service.ProductService;
+import org.springframework.stereotype.Service;
+
+@Service
+public class ProductServiceImpl implements ProductService {
+    private final ProductRepository productRepository;
+
+    public ProductServiceImpl(ProductRepository productRepository) {
+        this.productRepository = productRepository;
+    }
+
+    @Override
+    public Product save(Product product) {
+        return productRepository.save(product);
+    }
+
+    @Override
+    public Product findById(Long id) {
+        return productRepository.findById(id).orElseThrow(NoSuchElementException::new);
```

Comment:
check out the source code of `orElseThrow()` without params, it throws this exception for you, same thing in other service
```suggestion
        return productRepository.findById(id).orElseThrow();
```